### PR TITLE
Merge JavaScript and TypeScript sensors

### LINKED ISSUE #3728: Refactor the JavaScript sensor to reuse AnalysisWithProgram and AnalysisWithWatchProgram
To prepare the merging of both the JavaScript and TypeScript sensors, the JavaScript sensor should use the same implementation as the TypeScript sensor:
- `AnalysisWithProgram` when running in the scanner
- `AnalysisWithWatchProgram` when running in the SonarLint.

### LINKED ISSUE #3729: Merge the JavaScript and TypeScript sensors
Both sensors follow the same steps for JavaScript, TypeScript, and Vue files:

1. Search for TypeScript configuration files (`tsconfig.json` or `jsconfig.json`)
2. Map analyzed files to TypeScript configuration files
3. Rely on TypeScript ESLint for parsing using TypeScript `createProgram()` in the scanner and `createWatchProgram()` in SonarLint

As the NodeJS API supports multiple linters simultaneously through the linter id parameter, the new sensor can analyze JavaScript, TypeScript, and Vue files in one pass, improving the overall scanner performance.

### PR DESCRIPTION
**Pull request by saberduck**
Fixes #3729

### CODE REVIEW

**Review by victor-diez-sonarsource — APPROVED** [review]
LGTM to me after discussing with you!

File: sonar-plugin/sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/AbstractAnalysis.java [hunk 1]
```diff
@@ -36,14 +39,10 @@ abstract class AbstractAnalysis {
   final AnalysisProcessor analysisProcessor;
   SensorContext context;
   ContextUtils contextUtils;
-  AbstractChecks checks;
+  JsTsChecks checks;
```
[the remark this answers is not in the document]
Comment (saberduck) [reply sonar-plugin/sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/AbstractAnalysis.java:42]: indeed, I will fix this

File: sonar-plugin/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/TestUtils.java [hunk 2]
```diff
@@ -20,6 +20,10 @@
 package org.sonar.plugins.javascript;
 
 import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
```
[the remark this answers is not in the document]
Comment (saberduck) [reply sonar-plugin/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/TestUtils.java:25]: I guess because it's a test file

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
LGTM! Amazing job!

**Pull request merged**